src/jdk.compiler/share/classes/com/sun/tools/javac/util/Options.java

```diff
@@ -173,66 +173,119 @@ public boolean isUnset(Option option, String value) {
      * @param lc the {@link LintCategory} in question
      * @return true if {@code lc} has been disabled
      */
-    public boolean isLintDisabled(LintCategory lc) {
-        return isLintExplicitlyDisabled(lc) || isSet(Option.XLINT_CUSTOM, Option.LINT_CUSTOM_NONE);
+    public boolean isDisabled(Option option, LintCategory lc) {
+        return isExplicitlyDisabled(option, lc) || isSet(option.getCustom(), Option.LINT_CUSTOM_NONE);
     }
 
     /**
      * Determine if a specific {@link LintCategory} is explicitly enabled via a custom
-     * option flag of the form {@code -Xlint:key}.
+     * option flag of the form {@code -Flag:key}.
      *
      * <p>
-     * Note: This does not check for option flags of the form {@code -Xlint} or {@code -Xlint:all}.
+     * Note: This does not check for option flags of the form {@code -Flag} or {@code -Flag:all}.
      *
      * <p>
      * Note: It's possible the category was also disabled; this method does not check that.
      *
+     * @param option the plain (non-custom) version of the option (e.g., {@link Option#XLINT})
      * @param lc the {@link LintCategory} in question
      * @return true if {@code lc} has been explicitly enabled
      */
-    public boolean isLintExplicitlyEnabled(LintCategory lc) {
-        return lc.optionList.stream().anyMatch(alias -> isSet(Option.XLINT_CUSTOM, alias));
+    public boolean isExplicitlyEnabled(Option option, LintCategory lc) {
+        Option customOption = option.getCustom();
+        return lc.optionList.stream().anyMatch(alias -> isSet(customOption, alias));
     }
 
     /**
      * Determine if a specific {@link LintCategory} is explicitly disabled via a custom
-     * option flag of the form {@code -Xlint:-key}.
+     * option flag of the form {@code -Flag:-key}.
      *
      * <p>
-     * Note: This does not check for an option flag of the form {@code -Xlint:none}.
+     * Note: This does not check for an option flag of the form {@code -Flag:none}.
      *
      * <p>
      * Note: It's possible the category was also enabled; this method does not check that.
      *
+     * @param option the plain (non-custom) version of the option (e.g., {@link Option#XLINT})
      * @param lc the {@link LintCategory} in question
      * @return true if {@code lc} has been explicitly disabled
      */
-    public boolean isLintExplicitlyDisabled(LintCategory lc) {
-        return lc.optionList.stream().anyMatch(alias -> isSet(Option.XLINT_CUSTOM, "-" + alias));
+    public boolean isExplicitlyDisabled(Option option, LintCategory lc) {
+        Option customOption = option.getCustom();
+        return lc.optionList.stream().anyMatch(alias -> isSet(customOption, "-" + alias));
+    }
+
+    /**
+     * Collect the set of {@link LintCategory}s specified by option flag(s) of the form
+     * {@code -Flag} and/or {@code -Flag:[-]key,[-]key,...}.
+     *
+     * <p>
+     * The set of categories is calculated as folllows. First, an initial set is created:
```

Comment:
typo: `folllows`

Reply:
Thanks - should be fixed in e8f5539f3a1.